test/widgets/textfields_test.dart

```diff
@@ -57,4 +57,41 @@ void main() {
     await tester.pumpAndSettle();
     expect(find.text('entering 123 for cell field'), findsOneWidget);
   });
+
+  testWidgets('URL Field sends request on enter keystroke', (tester) async {
+    bool wasSubmitCalled = false;
+
+    void testSubmit(String val) {
+      wasSubmitCalled = true;
+    }
+
+    await tester.pumpWidget(
+      MaterialApp(
+        title: 'URL Field',
+        theme: kThemeDataDark,
+        home: Scaffold(
+          body: Column(children: [
+            URLField(
+              selectedId: '2',
+              onFieldSubmitted: testSubmit,
+            )
+          ]),
+        ),
+      ),
+    );
+
+    // ensure URLField is blank
+    expect(find.byType(TextFormField), findsOneWidget);
+    expect(find.textContaining('Enter API endpoint '), findsOneWidget);
+    expect(wasSubmitCalled, false);
+
+    // modify value and press enter
+    var txtForm = find.byKey(const Key("url-2"));
+    await tester.enterText(txtForm, 'entering 123');
+    await tester.testTextInput.receiveAction(TextInputAction.done);
```

Comment:
```suggestion
    await tester.sendKeyEvent(LogicalKeyboardKey.enter);
```
Since the user would most probably using a keyboard, this is a better event to test.

Reply:
Hey @mmjsmohit, thanks for your input. As mentioned in the docs, the action doesn't reach the flutter engine and stops at framework, causing the test to fail on your implementation.

See also: https://api.flutter.dev/flutter/widgets/EditableText/onSubmitted.html